src/effects/effectchainmanager.cpp

```diff
@@ -164,37 +168,58 @@ bool EffectChainManager::saveEffectChains() {
     return true;
 }
 
-bool EffectChainManager::loadEffectChains() {
-    //qDebug() << debugString() << "loadEffectChains";
+QList<std::pair<EffectChainPointer, QDomElement>> EffectChainManager::loadEffectChains() {
+    // StandardEffectRack::addEffectChainSlot uses both the EffectChainPointer
+    // and the saved state from the XML to initialize the respective
+    // EffectChain/Effect/EffectParameter/EffectButtonParameter Slots
+    QList<std::pair<EffectChainPointer, QDomElement>> loadedChains;
 
     QDir settingsPath(m_pConfig->getSettingsPath());
     QFile file(settingsPath.absoluteFilePath("effects.xml"));
+    QDomDocument doc;
 
     if (!file.open(QIODevice::ReadOnly)) {
-        return false;
+        EffectChainPointer pEmptyChain;
+        QDomElement emptyChainElement = doc.createElement("EffectChain");
+        for (int i = 0; i < 4; ++i) {
```

Comment:
Is this the same 4 we have in StandardEffectRack::addEffectChainSlot? 
Maybe it can become a const member of EffectChainManager,